### ISSUE TITLE
Can't delete_indices by monthly

### ISSUE THREAD

**Issue by dreampuf** [elastic/curator#897]
I have no idea why it adds a '1'.

```python
def get_datetime(index_timestamp, timestring):
    """
    Return the datetime extracted from the index name, which is the index
    creation time.

    :arg index_timestamp: The timestamp extracted from an index name
    :arg timestring: An strftime pattern
    :rtype: :py:class:`datetime.datetime`
    """
    # Compensate for week of year by appending '%w' to the timestring
    # and '1' (Monday) to index_timestamp
    if '%W' in timestring:
        timestring += '%w'
        index_timestamp += '1'
    elif '%U' in timestring:
        timestring += '%w'
        index_timestamp += '1'
    elif '%m' in timestring:
        if not '%d' in timestring:  #<---------This is what I concerned.
            timestring += '%d'
            index_timestamp += '1'
    return datetime.strptime(index_timestamp, timestring)
```

**Comment by untergeek** [comment]
It doesn't just add a 1, it adds a 1 to the timestamp and a `%d` to the timestring:

`timestring` becomes: `%Y.%m` -> `%Y.%m%d`
and
`index_timestamp` becomes: `2017.02` -> `2017.021`

```
>>> from datetime import datetime
>>> timestring = '%Y.%m'
>>> index_timestamp = '2017.02'
>>> datetime.strptime(index_timestamp, timestring)
datetime.datetime(2017, 2, 1, 0, 0)
>>> timestring += '%d'
>>> index_timestamp += '1'
>>> datetime.strptime(index_timestamp, timestring)
datetime.datetime(2017, 2, 1, 0, 0)
```

This admittedly silly workaround was added to support older versions of Python, as without it, some crufty old versions of datetime didn't parse it properly (I don't remember the PR that fixed this issue).  

In any case, it doesn't change the result in any way, so it shouldn't be acting as a block to you.

**Comment by untergeek** [comment]
This method also only returns the datetime timestamp of the local-only modifications.  It doesn't alter the index_timestamp or the timestring globally.

**Comment by dreampuf** [comment]
@untergeek Thank you explain. I got an error when I trying to delete monthly indices, which don't include the day unit. I will post the traceback late.

**Comment by untergeek** [comment]
Closing due to lack of follow up.  Feel free to reopen if you need to.

**Comment by dreampuf** [comment]
@untergeek I think I found the root cause. Some weekly indices are in the ES. When I execute this action, it will throw an error while the curator trying to parse the weekly indices.

I think I would change the name convention to identify which rule are they use.

**Comment by untergeek** [comment]
I think your issue is wholly resolved by using the [new `period` filter](https://www.elastic.co/guide/en/elasticsearch/client/curator/5.0/filtertype_period.html), available in Curator 5.0.
